### PR TITLE
Forum Help Channels

### PR DESCRIPTION
**Pull request by DynxstyGIT**
This PR implements the Help Channel XP & Help Channel Thanks functionalities into Discords' new [Forum Channels](https://support.discord.com/hc/en-us/articles/6208479917079-Forum-Channels-FAQ)
As far as I can tell, all other features from the text channel-based system are replaced by the forums' features, like auto archiving, slowmode on new posts, etc.

### CODE REVIEW

File: src/main/java/net/javadiscord/javabot/systems/help/forum/ForumHelpManager.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+						postThread.getIdLong(),
+						postThread.getOwnerIdLong(),
+						postThread.getIdLong(),
+						helper.getIdLong()
+				);
+				HelpConfig config = Bot.getConfig().get(event.getGuild()).getHelpConfig();
+				HelpExperienceService service = new HelpExperienceService(Bot.getDataSource());
+				// Perform experience transactions
+				service.performTransaction(helper.getIdLong(), config.getThankedExperience(), HelpTransactionMessage.GOT_THANKED, event.getGuild());
+				service.performTransaction(postThread.getOwnerIdLong(), config.getThankExperience(), HelpTransactionMessage.THANKED_USER, event.getGuild());
```
Comment (danthe1st) [inline src/main/java/net/javadiscord/javabot/systems/help/forum/ForumHelpManager.java:111]: Should the OP really get experience for every helper they thanked?

File: src/main/java/net/javadiscord/javabot/systems/help/forum/ForumHelpManager.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+						postThread.getAsMention(),
+						e.getMessage()
+				).queue();
+			}
+		});
+	}
+
+
+	private @NotNull List<Member> getPostHelpers() {
+		List<Member> helpers = new ArrayList<>(20);
```
Comment (danthe1st) [inline src/main/java/net/javadiscord/javabot/systems/help/forum/ForumHelpManager.java:127]: Streams could be used for this.

**Review by danthe1st — APPROVED** [review]

File: src/main/java/net/javadiscord/javabot/systems/help/forum/ForumHelpManager.java (hunk 1 above)
Comment (DynxstyGIT) [reply src/main/java/net/javadiscord/javabot/systems/help/forum/ForumHelpManager.java:111]: Well, it was always like that. Its just 3 XP, so, well. I think we can even get rid of that

File: src/main/java/net/javadiscord/javabot/systems/help/forum/ForumHelpManager.java (hunk 2 above)
Comment (DynxstyGIT) [reply src/main/java/net/javadiscord/javabot/systems/help/forum/ForumHelpManager.java:127]: sure

**Review by danthe1st — APPROVED** [review]

**Pull request merged**